lib/puppet/rest/client.rb

```diff
@@ -40,10 +42,14 @@ def initialize(route,
     # @param [String] endpoint the endpoint of the configured API to query
     # @param [Hash] query any URL params to add to send to the endpoint
     # @param [Hash] header any additional entries to add to the default header
-    # @return [Puppet::Rest::Response] the response from the server
-    def get(endpoint, query: nil, header: nil)
-      response = @client.get(endpoint, query: query, header: header)
-      Puppet::Rest::Response.new(response)
+    # @yield [HTTPClient::Message, String] the HTTP response and a chunk of its body
+    # @raise [Puppet::Rest::ResponseError] if the response status is not OK
+    def get(endpoint, query: nil, header: nil, &block)
+      begin
+        @client.get_content(endpoint, { query: query, header: header }, &block)
```

Comment:
Is it any better to call
```rb
@client.get_content(endpoint, {query: query, header: header}) do |response, chunk|
  block.call(response, chunk)
end
```
?

I had a moment when reading the docstring when I was like, "Wait, we don't yield anything though...."

I think maybe calling the upstream method in the way we advertise will mean if the calling conventions upstream change, we'll fail here, rather than more cryptically in our callers.

Reply:
We don't want to call the block here. The block is called several layers deeper by the HTTPClient library. See https://github.com/nahi/httpclient/blob/master/lib/httpclient.rb#L1250. We just want to pass it through to that method. I'm not sure what the best way to document this is.

Reply:
Looking at this more, I see what you mean here about calling the block -- my mind had parsed that as one level out, hence the confusion. I'll update it to do this for the reasons we talked about yesterday: clearer to see what's going on, easier to document, might help figuring out failures.